### PR TITLE
docs(samples): add more clustering code snippets

### LINKED ISSUE #329: Add more code snippets for creating clustered table.
Add sample code for **creating a clustered table from a query result**. 
Section: [https://cloud.google.com/bigquery/docs/creating-clustered-tables#creating_a_clustered_table_from_a_query_result](https://cloud.google.com/bigquery/docs/creating-clustered-tables#creating_a_clustered_table_from_a_query_result)

Add sample code for **creating a clustered table when you load data**.
Section: [https://cloud.google.com/bigquery/docs/creating-clustered-tables#creating_a_clustered_table_when_you_load_data](https://cloud.google.com/bigquery/docs/creating-clustered-tables#creating_a_clustered_table_when_you_load_data)

### PR DESCRIPTION
**Pull request by ryanyuan**
Add sample code for creating a clustered table from a query result.
File: samples/client_query_destination_table_clustered.py
Section: [https://cloud.google.com/bigquery/docs/creating-clustered-tables#creating_a_clustered_table_from_a_query_result](https://cloud.google.com/bigquery/docs/creating-clustered-tables#creating_a_clustered_table_from_a_query_result)

Add sample code for creating a clustered table when you load data.
File: samples/load_table_clustered.py
Section: [https://cloud.google.com/bigquery/docs/creating-clustered-tables#creating_a_clustered_table_when_you_load_data](https://cloud.google.com/bigquery/docs/creating-clustered-tables#creating_a_clustered_table_when_you_load_data)

Fixes #329  🦕

### CODE REVIEW

File: samples/client_query_destination_table_clustered.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def client_query_destination_table_clustered(table_id):
+
+    # [START bigquery_query_destination_table_clustered]
```
Comment (tswast) [inline samples/client_query_destination_table_clustered.py:18]: We need to standardize with what the other languages are using.

```suggestion
    # [START bigquery_query_clustered_table]
```

File: samples/client_query_destination_table_clustered.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    )
+
+    # Start the query, passing in the extra configuration.
+    query_job = client.query(sql, job_config=job_config)  # Make an API request.
+    query_job.result()  # Wait for the job to complete.
+
+    table = client.get_table(table_id)  # Make an API request.
+    if table.clustering_fields == cluster_fields:
+        print("The destination table is written using the cluster_fields configuration.")
+    # [END bigquery_query_destination_table_clustered]
```
Comment (tswast) [inline samples/client_query_destination_table_clustered.py:41]:
```suggestion
    # [END bigquery_query_clustered_table]
```

File: samples/load_table_clustered.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            bigquery.SchemaField("full_name", "STRING"),
+            bigquery.SchemaField("age", "INTEGER"),
+        ],
+        clustering_fields=["age"],
+        skip_leading_rows=1,
+        # The source format defaults to CSV, so the line below is optional.
+        source_format=bigquery.SourceFormat.CSV,
+    )
+
+    with open(file_path, "rb") as source_file:
```
Comment (tswast) [inline samples/load_table_clustered.py:38]: Nit: The Go sample loads from GCS `"gs://cloud-samples-data/bigquery/sample-transactions/transactions.csv"`

It'd be nice if we had some consistency across languages. Also, it's a little bit simpler to run the sample without the need for local files.

Go sample:

https://github.com/GoogleCloudPlatform/golang-samples/blob/7c9a079be0894b2c04c825c07d55176e65087c88/bigquery/snippets/loadingdata/bigquery_load_table_clustered.go#L17-L70

**Review by tswast — CHANGES_REQUESTED** [review]
Thanks for the contribution! A couple of nits

**Review by tswast — APPROVED** [review]

**Comment by ryanyuan** [comment]
@tswast thanks for the review and changes.

**Review by engelke — APPROVED** [review]
LGTM.

**Pull request merged**